lucene/core/src/test/org/apache/lucene/search/TestFieldSortOptimizationSkipping.java

```diff
@@ -290,5 +299,114 @@ public void testFloatSortOptimization() throws IOException {
     dir.close();
   }
 
+  public void testDocSortOptimizationWithAfter() throws IOException {
+    final Directory dir = newDirectory();
+    final IndexWriter writer = new IndexWriter(dir, new IndexWriterConfig());
+    final int numDocs = atLeast(1500);
+    for (int i = 0; i < numDocs; ++i) {
+      final Document doc = new Document();
+      writer.addDocument(doc);
+      if ((i > 0) && (i % 500 == 0)) {
+        writer.commit();
+      }
+    }
+    final IndexReader reader = DirectoryReader.open(writer);
+    IndexSearcher searcher = new IndexSearcher(reader);
+    final int numHits = 3;
+    final int totalHitsThreshold = 3;
+    final int searchAfter = 1400;
+
+    // sort by _doc with search after should trigger optimization
```

Comment:
you can try different values of  search after in a loop to increase the coverage?

Reply:
Addressed in 493d9eb